src/main/java/com/faforever/client/replay/LocalReplayVaultController.java

```diff
@@ -50,10 +51,15 @@ public void initialize() {
   protected void onDisplayDetails(ReplayBean replay) {
     JavaFxUtil.assertApplicationThread();
     replayDetailController.setReplay(replay);
+    replayDetailController.getRoot().addEventHandler(DeleteLocalReplayEvent.DELETE_LOCAL_REPLAY_EVENT_TYPE, event -> onLocalReplayDeleted());
```

Comment:
Use google eventbus, might need to be added as a dependency through spring injection and also registration added in the initialize function